vars/createGithubIssue.groovy

```diff
@@ -0,0 +1,67 @@
+def call(Map args = [:]){
+    def failureMessages = args.message
+    List<String> failedComponents = []
+
+    if (failureMessages.size() == 1 && failureMessages[0] == "Build failed") {
+        println("No component failed, skip creating github issue.")
+    }
+    else {
+        for(message in failureMessages.unique()){
+            java.util.regex.Matcher match = (message =~ /(?<=\bcomponent\s).*/)
+            String matched = match[0]
+            println(matched.split(" ")[0].trim())
+            failedComponents.add(matched.split(" ")[0].trim())
+        }
+
+        def yamlFile = readYaml(file: "manifests/${INPUT_MANIFEST}")
+        def currentVersion = yamlFile.build.version
+
+        for(component in yamlFile.components){
+            if (failedComponents.contains(component.name)) {
+                println("Component ${component.name} failed, creating github issue")
+                compIndex = failedComponents.indexOf(component.name)
+                create_issue(component.name, component.repository, currentVersion, failureMessages[compIndex])
+                sleep(time:3,unit:"SECONDS")
+            }
+        }
+    }
+}
+
+def create_issue(component, url, curVersion, failedMessage){
```

Comment:
can we please rename the `curVersion`.

Reply:
anything particular in mind? @prudhvigodithi

Reply:
We can go with `currentVersion` ?

Reply:
Done.